cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1087,15 +1088,24 @@ private boolean isValid(RunRecordDetail existing, ProgramRunStatus nextProgramSt
    *
    * @param limit count at most that many runs, stop if there are more.
    */
-  public int countActiveRuns(@Nullable Integer limit) throws IOException {
-    AtomicInteger count = new AtomicInteger(0);
+  public Map<ProgramRunStatus, AtomicInteger> countActiveRuns(@Nullable Integer limit) throws IOException {
+    Map<ProgramRunStatus, AtomicInteger> result = new ConcurrentHashMap<>();
     try (CloseableIterator<RunRecordDetail> iterator = queryProgramRuns(
       Range.singleton(getRunRecordNamespacePrefix(TYPE_RUN_RECORD_ACTIVE, null)),
       key -> !NamespaceId.SYSTEM.getNamespace().equals(key.getString(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD)),
       null, limit != null ? limit : Integer.MAX_VALUE)) {
-      iterator.forEachRemaining(m -> count.getAndIncrement());
+      iterator.forEachRemaining(m -> {
+        // if this is a program within a workflow or a preview run, counting is skipped.
+        if (m.getSystemArgs().containsKey(ProgramOptionConstants.WORKFLOW_NAME)
+          || Boolean.parseBoolean(
+          m.getSystemArgs().getOrDefault(ProgramOptionConstants.SKIP_PROVISIONING, "false"))) {
```

Comment:
Correct. If SKIP_PROVISIONING is set to true, we treat it as a preview run, thus skip counting. But we consider "false" as the default value so in case SKIP_PROVISIONING is not set, we simply count the active run.